File: CodingSamples/Java/Language/SubclassTest/src/SalesPerson.java
package payroll;

//SalesPerson is a subclass of Employee (super-class)
public class SalesPerson extends Employee {

	private double sales;

	public SalesPerson(int h, float r, double s) {
		super(h, r); //calling super-class constructor
		sales = s;
	}

	public double getSales() {
		return sales;
	}

	public void setSales(double value) {
		sales = value;
	}

	//method overriding - defining a method in a class whose declaration
	//matches with the declaration of a method in its super-class
	public double income() {
		double amount = super.income();
		if(sales >= 20000)
			amount += 0.05 * sales;
		return amount;
	}
}

File: CodingSamples/DotNet/Language/ClassTypeTest/DemoApp/Payroll/SalesPerson.cs
namespace Payroll
{
    //SalesPerson is a derived class of Employee
    //which is called its base class
    class SalesPerson : Employee
    {
        //automatic property
        public double Sales { get; set; }

        public SalesPerson(int h, float r, double s) : base(h, r)
        {
            Sales = s;
        }

        //method overriding - implementing a method in
        //a dervied class which is declared as virtual
        //in the base class with same return type and
        //same list of parameter types
        public override double Income()
        {
            double amount = base.Income();
            if(Sales >= 20000)
                amount += 0.05 * Sales;
            return amount;
        }

    }
}